synapse/lib/storm.py

```diff
@@ -1103,17 +1103,22 @@ def _stormOperNextSeq(self, query, oper):
     def _stormOperJoin(self, query, oper):
 
         args = oper[1].get('args')
-        opts = dict(oper[1].get('kwlist'))
-
-        dstp = args[0]
-        srcp = args[0]
+        if len(args) is 2:
+            srcp, dstp = args[0], args[1]
+        elif len(args) is 1:
+            srcp, dstp = args[0], args[0]
```

Comment:
So I am super rusty at reading source code...it looks like this says "if there is only one arg, set both 'src' and 'dst' to the value of that arg. This is different from "if there is only one arg, assume 'src' is the primary property of the inbound nodes", which is what pivot() does (and how we want join() to work). Or am I misunderstanding?

Reply:
Agreed.  join() should work just like pivot in the single argument case.